### DISCUSSION TITLE
What is exact difference between session and pairing at 'sign api' ?

### DISCUSSION THREAD

**Discussion by dovigod** [WalletConnect/walletconnect-monorepo#2360] [Developer Support] [answered]
Hi, I'm newby at wallet connect v2.

Currently, I'm having difficulties understanding difference between session pairing.

I'll be glab if anyone could help me understanding the difference :)

**Comment by chendinh** [comment]
Pairing is a topic encrypted by a symmetric key shared through a URI between two clients with the sole purpose of communicating session proposals

ok i will make it simple. pairing's expired time is longer than session, Dev will save pairing on local storage and use pairing to init socket again with new session. after we close application, session can be closed but paring. Paring is mother of session lol. I hope it help you understand easily.

**Reply by finessevanes** [reply, marked as answer]
Thanks for jumping in @chendinh. To add a little more context, sessions occur on a pairing. You can have multiple sessions on a single pairing. Here are some of the official definitions we have:
[Pairing](https://docs.walletconnect.com/2.0/advanced/glossary#pairing)
[Session](https://docs.walletconnect.com/2.0/advanced/glossary#session)

**Reply by nsaun-dq** [reply]
hey, @chendinh @finessevanes , 
1. I am working with react-native and I have never actually had a situation where I have to re-connect a session. even if the app is restarted.
2. also, can I have 2 sessions for one pairing existing at the same time?
3. let's suppose I connect to any dapp. it is a pairing. Then dapp sends session_proposal, I reject it. the pairing still exists, right? In this case should I also disconnect the pairing? what is the point of pairing when I have rejected the session?
